slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -65,6 +65,7 @@ trait MySQLProfile extends JdbcProfile { profile =>
     - SqlCapabilities.sequenceLimited
     - RelationalCapabilities.joinFull
     - JdbcCapabilities.nullableNoDefault
+    - JdbcCapabilities.distinguishesIntTypes
```

Comment:
@cvogt I removed the capability, which was causing the test to fail in [ModelBuilderTest.scala#L300](https://github.com/slick/slick/blob/51e14f2756ed29b8c92a24b0ae24f2acd0b85c6f/slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ModelBuilderTest.scala#L300) 
The alternative would have been to add a new supportsShort capability and have the capability removed from the MysqlProfile.

There also seem to be two test cases [ModelBuilderTest.scala#L327](https://github.com/slick/slick/blob/51e14f2756ed29b8c92a24b0ae24f2acd0b85c6f/slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ModelBuilderTest.scala#L327) commented out which return a different type and this seems to be a similar case.